resilience4j-hedge/src/main/java/io/github/resilience4j/hedge/HedgeConfig.java

```diff
@@ -0,0 +1,197 @@
+        this.windowSize = windowSize;
+        this.cutoff = cutoff;
+        this.hedgeExecutor = executor;
+    }
+
+    /**
+     * Returns a builder to create a custom HedgeConfig.
+     *
+     * @return a {@link HedgeConfig.Builder}
+     */
+    public static Builder custom() {
+        return new Builder();
+    }
+
+    public static Builder from(HedgeConfig baseConfig) {
+        return new Builder(baseConfig);
+    }
+
+    /**
+     * Creates a default Hedge configuration.
+     *
+     * @return a default Hedge configuration.
+     */
+    public static HedgeConfig ofDefaults() {
+        return new Builder().build();
+    }
+
+    public HedgeMetrics newMetrics() {
+        if (metricsType == MetricsType.PRECONFIGURED) {
+            return HedgeMetrics.ofPreconfigured(cutoff);
+        } else {
+            return HedgeMetrics.ofAveragePlus(shouldUseFactorAsPercentage, hedgeTimeFactor,
+                shouldMeasureErrors, windowSize);
+        }
+    }
+
+    public ScheduledExecutorService getHedgeExecutor() {
+        return hedgeExecutor;
+    }
+
+    @Override
+    public String toString() {
+        return "HedgeConfig{" +
+            shouldUseFactorAsPercentage + "," +
+            hedgeTimeFactor + "," +
+            shouldMeasureErrors + "," +
+            windowSize + "," +
+            cutoff +
+            "}";
+    }
+
+
+    public enum MetricsType {
+        PRECONFIGURED, AVERAGE_PLUS
+    }
+
+    public static class Builder {
+
+        private MetricsType metricsType = MetricsType.PRECONFIGURED;
+        private boolean shouldUseFactorAsPercentage = false;
+        private int hedgeTimeFactor = 0;
+        private boolean shouldMeasureErrors = true;
+        private int windowSize = 100;
+        private Duration cutoff;
+        private RejectedExecutionHandler rejectionHandler = new ThreadPoolExecutor.DiscardOldestPolicy();
+        private int concurrentHedges = 10;
+        private String threadsNamed = "HedgeThreadPool";
+        private ScheduledExecutorService providedExecutor;
+
+        public Builder() {
+        }
+
+        public Builder(HedgeConfig baseConfig) {
+            this.metricsType = baseConfig.metricsType;
+            this.shouldUseFactorAsPercentage = baseConfig.shouldUseFactorAsPercentage;
+            this.hedgeTimeFactor = baseConfig.hedgeTimeFactor;
+            this.shouldMeasureErrors = baseConfig.shouldMeasureErrors;
+            this.windowSize = baseConfig.windowSize;
+            this.cutoff = baseConfig.cutoff;
+            this.providedExecutor = baseConfig.hedgeExecutor;
+        }
+
+        public static Builder fromConfig(HedgeConfig baseConfig) {
+            return new Builder(baseConfig);
+        }
+
+        /**
+         * Builds a HedgeConfig
+         *
+         * @return the HedgeConfig
+         */
+        public HedgeConfig build() {
+            if (providedExecutor == null) {
+                providedExecutor = new ScheduledThreadPoolExecutor(
```

Comment:
I'd like that. The code should work fine when passed a ContextAwareScheduledThreadPoolExecutor. Given that this is a fallback where no executor has been passed, I am not sure what contextPropagators to pass it. I could pass the emptyPropagator, but that doesn't seem like it would add huge value?

Reply:
As a fallback it can be an `ContextAwareScheduledThreadPoolExecutor` with an empty array of context propagators. 
The `ContextAwareScheduledThreadPoolExecutor` atleast provides copying of MDC contexts.